thriftbp/server_middlewares.go

```diff
@@ -450,3 +453,50 @@ func PrometheusServerMiddleware(method string, next thrift.TProcessorFunction) t
 	}
 	return thrift.WrappedTProcessorFunction{Wrapped: process}
 }
+
+// ServerHeaderBPMiddleware is a middleware that extracts baseplate headers from the incoming request and adds them to the context.
+func ServerHeaderBPMiddleware() thrift.ProcessorMiddleware {
+	return func(name string, next thrift.TProcessorFunction) thrift.TProcessorFunction {
+		return thrift.WrappedTProcessorFunction{
+			Wrapped: func(ctx context.Context, seqID int32, in, out thrift.TProtocol) (bool, thrift.TException) {
+				readHeaderList := thrift.GetReadHeaderList(ctx)
+				if len(readHeaderList) == 0 {
+					return next.Process(ctx, seqID, in, out)
+				}
+
+				// check both the lower case and the http canonical case since thrift.GetHeader is case sensitive
+				var untrusted bool
+				if _, ok := thrift.GetHeader(ctx, headerbp.IsUntrustedRequestHeaderLower); ok {
+					untrusted = true
+				} else if _, ok := thrift.GetHeader(ctx, headerbp.IsUntrustedRequestHeaderCanonicalHTTP); ok {
+					untrusted = true
+				}
+				if untrusted {
```

Comment:
I wonder if we should add a `x-bp-` version of the deadline propagation header, so that we can still propagate deadline even in a untrusted chain of calls?

Reply:
> I wonder if we should add a x-bp- version of the deadline propagation header, so that we can still propagate deadline even in a untrusted chain of calls?

I'm not sure I understand, this specifically would discard that header if the request was marked as untrusted? 😅